components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -265,10 +266,15 @@ class SitePermissionsFeature(
 
     private fun handleRuledFlow(
         permissionRequest: PermissionRequest,
-        session: Session
+        session: Session,
+        sitePermissionsRules: SitePermissionsRules
     ): SitePermissionsDialogFragment? {
-        val action = requireNotNull(sitePermissionsRules).getActionFrom(permissionRequest)
+        val action = sitePermissionsRules.getActionFrom(permissionRequest)
         return when (action) {
+            SitePermissionsRules.Action.ALLOWED -> {
+                permissionRequest.grant()
```

Comment:
We need to consume the request after granting   `session.contentPermissionRequest.consume { true }`